kotlin-ktor/src/test/java/com/baeldung/graphql/client/GraphQLClientTest.kt

```diff
@@ -0,0 +1,142 @@
+package com.baeldung.graphql.client
+
+import com.baeldung.graphql.server.data.AttendeeRepository
+import com.baeldung.graphql.server.data.ConferenceRepository
+import com.baeldung.graphql.server.graphQLModule
+import io.ktor.server.engine.*
+import io.ktor.server.netty.*
+import kotlinx.coroutines.runBlocking
+import org.junit.AfterClass
+import org.junit.BeforeClass
+import kotlin.test.*
+
+class GraphQLClientTest {
```

Comment:
In some of these tests, we only assert on the content of the repository, but do not check the result of the client call itself. This is particularly important in the case where we create with a null id, and then the client might rely on that id for future lookups.

For those cases, can we do something along the lines of:
```
        val response = graphQLService.createOrSaveAttendee(null, "Some Attendee")
        assertEquals(0, response?.id)
        assertEquals("Some Attendee", response?.name)

        val repository = AttendeeRepository.findById(0)
        assertEquals("Some Attendee", repository.name)
```

To make it clear that there is consistency between the repository and the response returned to the client.

Reply:
The thing is, I wanted to avoid this for all the tests as it really clutters the readability of the tests and we start testing out-of-scope behavior in tests meant to check just one part of the functionality. I think this makes sense with the tests that cover creation, but not so sure about the other tests. What do you think?

Reply:
Yes agreed as long as it is covered once somewhere, checks elsewhere can be looser